lib/ide/ast/impl/src/lib.rs

```diff
@@ -152,6 +152,28 @@ impl Ast {
         self
     }
 
+    // TODO use strongly typed IDMap, after it's merged to master
+
+    fn collect_ids(&self, index:&mut usize, ids:&mut IDMap) {
+        self.id.map( |id| {
+            ids.push(((*index, self.span), id));
+            *index += self.span;
+        });
+
+        for child in self {
+            child.collect_ids(index, ids);
+        }
+    }
+
+    pub fn id_map(&self) -> IDMap {
```

Comment:
No docs - please be sure that all lints are enabled in this codebase (lints like `warn(missing_docs)`). Ask @mwu-tow for help in case of doubt.

Reply:
@Josef-Vonasek Take a look at https://github.com/luna/ide/blob/master/lib/ide/utils/src/lib.rs

Unfortunately we have enable lints manually in main file of each library crate.